components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -0,0 +1,468 @@
+					subWithType.Spec.Config[v1alpha2.MaxInFlightMessages] = "20"
+					jsCtx.On("ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName()).Return(defaultConsumerInfo, nil)
+					jsCtx.On("UpdateConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig")).Return(nil, nats.ErrJetStreamNotEnabled)
+				}
+			},
+			givenAssertions: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				// check if the expected methods got called
+				for _, eventType := range sub.Status.Types {
+					jsSubject := jsBackend.getJetStreamSubject(sub.Spec.Source, eventType.CleanType, sub.Spec.TypeMatching)
+					jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
+
+					jsCtx.AssertCalled(t, "ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName())
+
+					defaultConsumerInfo.Config.MaxAckPending = 20
+					jsCtx.AssertCalled(t, "UpdateConsumer", jsBackend.Config.JSStreamName, &defaultConsumerInfo.Config)
+					jsCtx.AssertNotCalled(t, "AddConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig"))
+				}
+			},
+			wantErr: errorMessage(nats.ErrJetStreamNotEnabled.Error()),
+		},
+	}
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			testEnv := getTestEnvironmentWithMock(t)
+			if testCase.givenSubscriptionMap != nil {
+				testEnv.jsBackend.subscriptions = testCase.givenSubscriptionMap
+			}
+			for _, eventType := range testCase.givenSubscription.Status.Types {
+				jsSubject := testEnv.jsBackend.getJetStreamSubject(testCase.givenSubscription.Spec.Source, eventType.CleanType, testCase.givenSubscription.Spec.TypeMatching)
+				jsSubKey := NewSubscriptionSubjectIdentifier(testCase.givenSubscription, jsSubject)
+
+				// mock the expected calls
+				tc.givenMocks(testCase.givenSubscription, testEnv.jsBackend, testEnv.jsCtxMock, jsSubKey)
+			}
+
+			// when
+			err := testEnv.jsBackend.syncNATSConsumers(testCase.givenSubscription, callback, testEnv.logger.WithContext())
+
+			// then
+			// check if error is expected
+			if testCase.wantErr == nil {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				require.Equal(t, err.Error(), *testCase.wantErr)
+			}
+
+			// then
+			// check the expected assetions
+			for _, eventType := range tc.givenSubscription.Status.Types {
+				jsSubject := testEnv.jsBackend.getJetStreamSubject(testCase.givenSubscription.Spec.Source, eventType.CleanType, testCase.givenSubscription.Spec.TypeMatching)
+				jsSubKey := NewSubscriptionSubjectIdentifier(testCase.givenSubscription, jsSubject)
+
+				tc.givenAssertions(testCase.givenSubscription, testEnv.jsBackend, testEnv.jsCtxMock, jsSubKey)
+			}
+		})
+	}
+}
+
+// Test_CheckNATSSubscriptionsCount test the behaviour of the checkNATSSubscriptionsCount function.
+func Test_CheckNATSSubscriptionsCount(t *testing.T) {
```

Comment:
I tried to simplify the test. The details are in this [PR](https://github.com/VladislavPaskar/kyma/pull/426). Let me know what you think or we can discuss it in a call as well 😃

Reply:
I like it, thanks for contribution